src/Tests/Microsoft.NET.Publish.Tests/PublishItemsOutputGroupTests.cs

```diff
@@ -144,7 +144,47 @@ public void GroupPopulatedCorrectlyWithSingleFile()
             }
         }
 
-        private TestProject SetupProject()
+        [CoreMSBuildAndWindowsOnlyFact]
+        public void GroupBuildsWithoutPublish()
+        {
+            var testProject = this.SetupProject(addCopyFilesTargets: false);
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var restoreCommand = new RestoreCommand(Log, testAsset.Path, testProject.Name);
+            restoreCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var buildCommand = new BuildCommand(Log, testAsset.Path, testProject.Name);
+            buildCommand
+                .Execute("/p:RuntimeIdentifier=win-x86;DesignTimeBuild=true;PublishSingleFile=true", "/t:PublishItemsOutputGroup")
+                .Should()
+                .Pass();
+        }
+
+        [CoreMSBuildAndWindowsOnlyFact]
+        public void GroupBuildDoesNotGeneratePublishFiles()
+        {
+            var testProject = this.SetupProject();
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var restoreCommand = new RestoreCommand(Log, testAsset.Path, testProject.Name);
+            restoreCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var buildCommand = new BuildCommand(Log, testAsset.Path, testProject.Name);
+
+            // Since this build tries to copies all the final publish artifacts without calling the publish target it should fail
+            buildCommand
+                .Execute("/p:RuntimeIdentifier=win-x86;DesignTimeBuild=true;PublishSingleFile=true", "/t:PublishItemsOutputGroup")
```

Comment:
Would it make more sense just to check that there's nothing in the publish folder after running the PublishItemsOutputGroup target (but not Publish)?

Also you may be able to use the GetValuesCommand to gather the files from the output group in a list instead of having to write your own target to copy them and then check the contents.

Reply:
I second this.

Reply:
Good point, I can just verify the build succeeds without the publish dir being created.  Updated.

I also see what you mean with GetValuesCommand.  I didn't know it existed when I first wrote these tests so I added the copy files targets in order to verify the contents of the output groups.  I'll leave this as is for now in this release branch change, but will refactor to use this next time these are updated.